control-plane/connect-inject/endpoints_controller.go

```diff
@@ -446,21 +334,37 @@ func (r *EndpointsController) createServiceRegistrations(pod corev1.Pod, service
 	}
 	tags := consulTags(pod)
 
```

Comment:
@mkeeler would love if you could take a look at this function. Here it is without the diff to make it a bit easier to read: https://github.com/hashicorp/consul-k8s/blob/53a26e2f2206fbb782b97e5ce1bc6266474bfada/control-plane/connect-inject/endpoints_controller.go#L293

For context, this is the function that will get called every time when there's any update to the kubernetes service's endpoints (e.g. when a new service is created or when a new pod added etc). 

My main question is about node registration. I think you mentioned before that we'd need to register the node before we register services. Currently in this code, I didn't do this but still provided `SkipNodeUpdate: true` and all tests (unit and acceptance) seem to be happy. I'm wondering if I'm missing something.

Let me know if this is enough context to review!

Reply:
`SkipNodeUpdate` only has an effect when the node is already registered. When its not already registered Consul will still register the node. This sort of makes sense because its a create and not an update that is happening.

I suspect your tests pass because they are either not using ACLs or the token used has the necessary `node:write` privileges.